packages/merge-on-green/src/merge-on-green.ts

```diff
@@ -213,30 +224,102 @@ handler.checkIfPRIsInvalid = async function checkIfPRIsInvalid(
   }
 };
 
+/**
+ * Checks if a branch with a PR has branch protection, if not, comments on PR
+ * @param owner type string
+ * @param repo type string
+ * @param prNumber type number
+ * @param github type githup API surface from payload
+ */
+handler.checkForBranchProtection = async function checkForBranchProtection(
+  owner: string,
+  repo: string,
+  prNumber: number,
+  github: InstanceType<typeof ProbotOctokit>
```

Comment:
nit, you could potentially do this once by going:

```
type GitHubType = InstanceType<typeof ProbotOctokit>
```

towards the top of the file.